File: src/main/java/javabot/operations/time/TimezonesAustralia.java
package javabot.operations.time;

public class TimezonesAustralia implements Timezones {

    private static Tri<String> timezones = new Tri<String>( );

    static {
        for( AustralianState states: AustralianState.values( ) ) {
            String timezone = states.getTimezone( );
            timezones.insert( states.name( ).toLowerCase( ), timezone );
            timezones.insert( states.getAbbreviation( ).toLowerCase( ), timezone );
            timezones.insert( states.getCapital( ), timezone );
        }
    }

    public enum AustralianState {
        WESTERN_AUSTRALIA( 8, "WA", "Perth", "Australia/Perth" ),
        SOUTH_AUSTRALIA( 5, "SA", "Adelaide", "Australia/Adelaide" ),
        NORTH_TERRITORY( 3, "NT", "Darwin", "Australia/Darwin" ),
        QUEENSLAND( 4, "QLD", "Brisbane", "Australia/Brisbane" ),
        NEW_SOUTH_WALES( 2, "NSW", "Sydney", "Australia/Sydney" ),
        AUSTRALIAN_CAPITAL_TERRITORY( 1, "ACT", "Canberra", "Australia/Canberra"  ),
        VICTORIA( 7, "VIC", "Melbourne", "Australia/Victoria" ),
        TASMANIA( 6, "TAS", "Hobart", "Australia/Hobart" );

        private final String timezone;
        private final String capital;
        private final String abbreviation;

        AustralianState( final int regionCode, final String abbreviation, final String capital,
                final String timezone ) {
            this.abbreviation = abbreviation;
            this.timezone = timezone;
            this.capital = capital;
        }

        public String getAbbreviation( ) {
            return abbreviation;
        }

        public String getCapital( ) {
            return capital;
        }

        public String getTimezone( ) {
            return timezone;
        }

    }

    @Override
    public String get( String province ) {
        return timezones.get( province );
    }

    @Override
    public String getCapital( ) {
        return AustralianState.NEW_SOUTH_WALES.getTimezone( );
    }
}

File: src/main/kotlin/javabot/operations/time/TimezonesAustralia.kt
package javabot.operations.time

class TimezonesAustralia : Timezones {

    enum class AustralianState(
        val abbreviation: String,
        val capital: String,
        val timezone: String
    ) {
        WESTERN_AUSTRALIA("WA", "Perth", "Australia/Perth"),
        SOUTH_AUSTRALIA("SA", "Adelaide", "Australia/Adelaide"),
        NORTH_TERRITORY("NT", "Darwin", "Australia/Darwin"),
        QUEENSLAND("QLD", "Brisbane", "Australia/Brisbane"),
        NEW_SOUTH_WALES("NSW", "Sydney", "Australia/Sydney"),
        AUSTRALIAN_CAPITAL_TERRITORY("ACT", "Canberra", "Australia/Canberra"),
        VICTORIA("VIC", "Melbourne", "Australia/Victoria"),
        TASMANIA("TAS", "Hobart", "Australia/Hobart")
    }

    override fun get(province: String): String? {
        return timezones.get(province)
    }

    override fun getCapital(): String {
        return AustralianState.NEW_SOUTH_WALES.timezone
    }

    companion object {

        private val timezones = Tri<String>()

        init {
            for (states in AustralianState.values()) {
                val timezone = states.timezone
                timezones.insert(states.name.toLowerCase(), timezone)
                timezones.insert(states.abbreviation.toLowerCase(), timezone)
                timezones.insert(states.capital, timezone)
            }
        }
    }
}
